lib/metasploit/framework/login_scanner/syncovery_file_sync_backup.rb

```diff
@@ -0,0 +1,139 @@
+        def linux?
+          globals = normalize_uri("#{uri}/get_global_variables")
+          res = send_request({ 'uri' => globals })
+          if res && res.code == 200
+            if res.body.scan(/"isSyncoveryLinux":"true"/).flatten[0] || res.body.scan(/"isSyncoveryWindows":"false"/).flatten[0]
+              return true
+            end
+
+            false
+          end
+          false
+        end
+
+        # Gets the Syncovery version.
+        #
+        # @return [String] version if version was found, otherwise FalseClass
+        def get_version
+          globals = normalize_uri("#{uri}/get_global_variables")
+          res = send_request({ 'uri' => globals })
+
+          if res && res.code == 200
+            version = res.body.scan(/"SyncoveryTitle":"Syncovery\s([A-Za-z0-9.]+)/).flatten[0] || ''
+            return version
+          end
+
+          false
+        end
+
+        # Actually doing the login. Called by #attempt_login
+        #
+        # @param username [String] The username to try
+        # @param password [String] The password or token to try
+        # @return [Hash]
+        #   * :status [Metasploit::Model::Login::Status]
+        #   * :proof [String] the HTTP response body or the session token
+        def get_login_state(username, password)
+          # Prep the data needed for login
+          if username.empty?
+            # no username => token is used as password
+            res = send_request({
+              'uri' => normalize_uri("#{uri}/profiles.json"),
+              'vars_get' => {
+                'recordstartindex' => '0',
+                'recordendindex' => '0'
+              },
+              'method' => 'GET',
+              'headers' => {
+                'token' => password
+              }
+            })
+            unless res
+              return { status: LOGIN_STATUS::UNABLE_TO_CONNECT, proof: res.to_s }
+            end
+            if !res.body.to_s.include? 'Session Expired'
+              return { status: LOGIN_STATUS::SUCCESSFUL, proof: res.body.to_s }
+            end
+
+            return { status: LOGIN_STATUS::INCORRECT, proof: res.body.to_s }
+          else
+            # use username:password
+            res = send_request({
+              'uri' => normalize_uri("#{uri}/post_applogin.php"),
+              'vars_get' => {
+                'login' => username.to_s,
+                'password' => password.to_s
+              },
+              'method' => 'GET'
+            })
+            unless res
+              return { status: LOGIN_STATUS::UNABLE_TO_CONNECT, proof: res.to_s }
+            end
+
+            # After login, the application should give us a new token
+            # session_token is actually just base64(MM/dd/yyyy HH:mm:ss) at the time of the login
+            token = res.body.scan(%r{"session_token":"((?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?)"}).flatten[0] || ''
```

Comment:
Not a blocker, framework has a `get_json_document` method that can parse JSON for you. That way if the token format ever changes in the future, this should continue to work without needing to change a regex

```suggestion
            json_res = res.get_json_document
            unless json_res
               # ....
            end
            
            token = json_res['session_token']
```

Reply:
It be worthwhile using `get_json_document` in the the other places that use regex too, just incase the JSON format is ever modified in the future - i.e. if they add spaces between the key and value etc.